### PR TITLE
[llvm-readelf] Update outdated URL

### PR DESCRIPTION
**Pull request by wenzhoumei**
This updates a comment pointing to the list of registered machine architectures in the ELF gABI as the URL in the old comment is no longer valid.

### CODE REVIEW

**Review by jh7370 — COMMENTED** [review]
It's worth noting that this isn't actually the latest set of registered EM_* values, just the latest set documented on the sco website. Unfortunately, they don't actively maintain/publish revisions to the ELF gABI. Instead, updates can be found on the ELF generic-abi mailing list.

Specifically, at the time of writing, I believe the latest list can be found at https://groups.google.com/g/generic-abi/c/0kORSDcyhTE/m/ZRf_PvcHAAAJ. You'll note there are a number listed there which aren't in the latest published ABI.

**Comment by wenzhoumei** [comment]
Hi @jh7370, thanks for the response! Would you recommend that I leave it as is or point to the general link of the mailing list instead (https://groups.google.com/g/generic-abi).

Also I'm very new to this so I'm not sure why tests are failing and what to do in this scenario. Will merging my branch on the latest version fix it?

**Comment by jh7370** [comment]
> Hi @jh7370, thanks for the response! Would you recommend that I leave it as is or point to the general link of the mailing list instead (https://groups.google.com/g/generic-abi).

As the point of the link is to provide people with a reference point for the registered machine constants, I'd be inclined to include both the link from my previous comment and the generic mailing list link, with some updates to the comment. Something along the lines that "At the time of writing, the list of registered machine architectures is at <link>. Please refer to <generic ABI mailing list> for any further updates."

> Also I'm very new to this so I'm not sure why tests are failing and what to do in this scenario. Will merging my branch on the latest version fix it?

Taking a quick look at the failures shows that these tests aren't anything to do with your change (which was obvious anyway, since you've only touched a comment). This means you can safely ignore them. They'll be broken in `main` at the time you created your branch and will have either been fixed by the time this gets merged, or will be fixed in the future. Either way, you don't need to do anything special. If you're fussed, you could try making a merge commit from main into your branch and pushing that to see if the test is fixed.

File: llvm/include/llvm/BinaryFormat/ELF.h [hunk 1]
```diff
@@ -127,9 +127,11 @@ enum {
...
 enum { EV_NONE = 0, EV_CURRENT = 1 };
 
-// Machine architectures
-// See current registered ELF machine architectures at:
-//    http://www.uxsglobal.com/developers/gabi/latest/ch4.eheader.html
+// Machine architectures.
+// At the time of writing, the list of registered machine architectures is
+// at https://groups.google.com/g/generic-abi/c/0kORSDcyhTE/m/ZRf_PvcHAAAJ
+// Please refer to https://groups.google.com/g/generic-abi for any further
+// updates
```
Comment (jh7370) [inline llvm/include/llvm/BinaryFormat/ELF.h:134]: Nit: please add a trailing "." here - LLVM comments should be full sentences and always end in a period, per the style guide.

**Review by jh7370 — APPROVED** [review]
LGTM. Do you need me to merge this for you?

**Comment by wenzhoumei** [comment]
> LGTM. Do you need me to merge this for you?

Yes please

**Comment by jh7370** [comment]
Hi @wenzhoumei, I went to merge this and I noticed that your email address is set to private in GitHub. The [LLVM Developer Policy](https://llvm.org/docs/DeveloperPolicy.html#making-and-submitting-a-patch) states the following:

> The LLVM project uses email to communicate to contributors outside of the GitHub platform about their past contributions. Primarily, our buildbot infrastructure uses emails to contact contributors about build and test failures. Therefore, the LLVM community requires contributors to have a public email address associated with their GitHub commits, so please ensure that “Keep my email addresses private” is disabled in your [account settings](https://github.com/settings/emails).

Please could you adjust the setting and let me know when you have, so that I can merge this change. Thanks!

**Comment by wenzhoumei** [comment]
Oh sorry @jh7370 , it should be set to public now. Thanks for the help!

**Comment by jh7370** [comment]
> Oh sorry @jh7370 , it should be set to public now. Thanks for the help!

No problem, it's easy to miss. All sorted.

**Pull request merged**